src/main/java/edu/ufpr/cluster/algorithms/functions/impl/JoinClustersFunction.java

```diff
@@ -2,40 +2,44 @@
 
 import com.google.common.collect.Lists;
 
+import org.uma.jmetal.util.pseudorandom.JMetalRandom;
+
 import edu.ufpr.cluster.algorithm.Cluster;
 import edu.ufpr.cluster.algorithm.ClusteringContext;
+import edu.ufpr.cluster.algorithm.Point;
 import edu.ufpr.cluster.algorithms.functions.Function;
-import edu.ufpr.cluster.random.ClusteringRandom;
 
 public class JoinClustersFunction implements Function<ClusteringContext> {
 
 	@Override
 	public void apply(ClusteringContext context) {
 
-		if (context.getClusters().size() > 2) {
+		if (context.getClusters().size() <= 1)
+			return;
 
-			int r = ClusteringRandom.getInstance().nextInt(0, context.getClusters().size() - 1);
+		int r = JMetalRandom.getInstance().nextInt(0, context.getClusters().size() - 1);
```

Comment:
We should use the ClusteringRandom inside the functions. To be able to control the seeds in our experiments